sympy/sets/fancysets.py

```diff
@@ -446,20 +456,20 @@ def __new__(cls, *args):
             raise ValueError("step cannot be 0")
 
         start, stop, step = slc.start or 0, slc.stop, slc.step or 1
-        try:
-            start, stop, step = [
-                w if w in [S.NegativeInfinity, S.Infinity]
-                else sympify(as_int(w))
-                for w in (start, stop, step)]
-        except ValueError:
-            raise ValueError(filldedent('''
-    Finite arguments to Range must be integers; `imageset` can define
-    other cases, e.g. use `imageset(i, i/10, Range(3))` to give
-    [0, 1/10, 1/5].'''))
 
-        if not step.is_Integer:
+        start, stop, step = map(_sympify, [start, stop, step])
+
+        for w in (start, stop, step):
```

Comment:
if Tuple(start, stop, step).atoms(Float):
    raise ValueError('Floats are not permitted in Range')